src/user-pool.ts

```diff
@@ -0,0 +1,233 @@
+} = {
+  createAuthChallenge: aws_cognito.UserPoolOperation.CREATE_AUTH_CHALLENGE,
+  customEmailSender: aws_cognito.UserPoolOperation.CUSTOM_EMAIL_SENDER,
+  customMessage: aws_cognito.UserPoolOperation.CUSTOM_MESSAGE,
+  customSmsSender: aws_cognito.UserPoolOperation.CUSTOM_SMS_SENDER,
+  defineAuthChallenge: aws_cognito.UserPoolOperation.DEFINE_AUTH_CHALLENGE,
+  postAuthentication: aws_cognito.UserPoolOperation.POST_AUTHENTICATION,
+  postConfirmation: aws_cognito.UserPoolOperation.POST_CONFIRMATION,
+  preAuthentication: aws_cognito.UserPoolOperation.PRE_AUTHENTICATION,
+  preSignUp: aws_cognito.UserPoolOperation.PRE_SIGN_UP,
+  preTokenGeneration: aws_cognito.UserPoolOperation.PRE_TOKEN_GENERATION,
+  userMigration: aws_cognito.UserPoolOperation.USER_MIGRATION,
+  verifyAuthChallengeResponse:
+    aws_cognito.UserPoolOperation.VERIFY_AUTH_CHALLENGE_RESPONSE,
+};
+
+export interface UserPoolProps
+  extends Omit<aws_cognito.UserPoolProps, "lambdaTriggers"> {
+  /**
+   * Lambda functions to use for supported Cognito triggers.
+   * @see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools-working-with-aws-lambda-triggers.html
+   * @default - No Lambda triggers.
+   */
+  readonly lambdaTriggers?: UserPoolTriggers;
+}
+
+/**
+ * Defines an AWS Cognito User Pool.
+ */
+export class UserPool {
+  public static readonly FunctionlessType = "UserPool";
+  public readonly FunctionlessType = UserPool.FunctionlessType;
+
+  /**
+   * The underlying {@link aws_cognito.UserPool} instance.
+   */
+  readonly resource: aws_cognito.UserPool;
+
+  /**
+   * Create a new {@link UserPool} from an existing {@link aws_cognito.UserPool}.
+   *
+   *
+   * @param resource an existing UserPool Construct.
+   * @see {@link aws_cognito.UserPool}
+   */
+  constructor(resource: aws_cognito.UserPool);
+
+  /**
+   * Create a new AWS Cognito {@link UserPool}.
+   *
+   * @param scope Construct scope to instantiate the UserPool in.
+   * @param id id of this UserPool within
+   * @param props the {@link UserPoolProps}.
+   * @see {@link aws_cognito.UserPool}
+   */
+  constructor(scope: Construct, id: string, props?: UserPoolProps);
+
+  constructor(
+    ...args:
+      | [resource: aws_cognito.UserPool]
+      | [scope: Construct, id: string, props?: UserPoolProps]
+  ) {
+    if (args.length === 1) {
+      this.resource = args[0];
+    } else {
+      const [scope, id, props] = args;
+      this.resource = new aws_cognito.UserPool(scope, id, {
+        ...props,
+        lambdaTriggers: props?.lambdaTriggers
+          ? Object.fromEntries(
+              Object.entries(props.lambdaTriggers).map(
+                ([triggerName, triggerHandler]) => [
+                  triggerName,
+                  triggerHandler.resource,
+                ]
+              )
+            )
+          : undefined,
+      });
+    }
+  }
+
+  /**
+   * Configure a {@link Function} to handle AWS Cognito Triggers.
```

Comment:
Nit: `Configure AWS Cognito to trigger a {@link Function}.` (The function is usable in theory).